runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -54,7 +57,11 @@
   "rawtypes", // TODO(https://github.com/apache/beam/issues/20447)
 })
 class DataflowMetrics extends MetricResults {
+
   private static final Logger LOG = LoggerFactory.getLogger(DataflowMetrics.class);
+  // TODO (rosinha): Remove this once bounded_trie is available in metrics proto Dataflow
```

Comment:
I would be more comfortable excluding these changes from this CL until the dataflow client protos are updated. We can do this in a follow-up PR (this one is already huge enough) rather than have a dummy implementation here (especially given the issues we had last time with class cast exceptions that suddenly manifest when the service is updated). The most important is that such metrics are reported and the round trip works on direct runners.